File: src/main/java/com/example/demo/controller/BannerController.java
package com.example.demo.controller;

import com.example.demo.entity.Banner;
import com.example.demo.service.IBannerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
@RestController
public class BannerController {
    Map<String,Object>bannMap=new HashMap<>();
    @Autowired
    private IBannerService bannerService;

    @RequestMapping(value = "USR000100002")
    public Object getBanner(Banner banner){
        List<Banner> result= bannerService.GetBannerData(banner);
        if (result.size()>0){
            bannMap.put("rescode","000000");
        } else {
            bannMap.put("rescode","999999");
        }
        bannMap.put("resobj",result);
        return bannMap;
    }
}

File: src/main/java/com/example/demo/controller/BannerController.kt
package com.example.demo.controller

import com.example.demo.entity.ApiResponse
import com.example.demo.entity.Banner
import com.example.demo.service.BannerService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class BannerController {
    @Autowired
    private var bannerService: BannerService? = null

    @RequestMapping(value = ["USR000100002"])
    fun getBanner(banner: Banner): ApiResponse<List<Banner>> {
        if (null == bannerService){
            return ApiResponse.error(null)
        }
        val result = bannerService?.bannerList(banner)
        return ApiResponse.success(data = result)
    }
}
